packages/jaeger-ui/src/utils/config/get-config.tsx

```diff
@@ -26,36 +26,47 @@ let haveWarnedDeprecations = false;
  * default config from `../../constants/default-config`.
  */
 const getConfig = memoizeOne(function getConfig() {
-  const getJaegerUiConfig = window.getJaegerUiConfig;
-  if (typeof getJaegerUiConfig !== 'function') {
-    if (!haveWarnedFactoryFn) {
-      // eslint-disable-next-line no-console
-      console.warn('Embedded config not available');
-      haveWarnedFactoryFn = true;
-    }
-    return { ...defaultConfig };
-  }
-  const embedded = getJaegerUiConfig();
-  if (!embedded) {
-    return { ...defaultConfig };
-  }
+  const embedded = getUiConfig();
+  const capabilities = getCapabilities();
   // check for deprecated config values
   if (Array.isArray(deprecations)) {
     deprecations.forEach(deprecation => processDeprecation(embedded, deprecation, !haveWarnedDeprecations));
     haveWarnedDeprecations = true;
   }
-  const rv = { ...defaultConfig, ...embedded };
+  const rv = { ...defaultConfig, ...embedded, ...capabilities };
   // mergeFields config values should be merged instead of fully replaced
   const keys = mergeFields;
   for (let i = 0; i < keys.length; i++) {
     const key = keys[i];
-    if (typeof embedded[key] === 'object' && embedded[key] !== null) {
+    if (embedded && typeof embedded[key] === 'object' && embedded[key] !== null) {
       rv[key] = { ...defaultConfig[key], ...embedded[key] };
     }
   }
   return rv;
 });
 
+function getUiConfig() {
+  const getJaegerUiConfig = window.getJaegerUiConfig;
+  if (typeof getJaegerUiConfig !== 'function') {
+    if (!haveWarnedFactoryFn) {
```

Comment:
I wonder if we even need this check given that the whole top-level `getConfig` function is memoized

Reply:
Do you have any idea about the behavior if the function is called while the backend has not injected the functions into the index.html? Or even if this scenario is likely to happen?

Reply:
I don't think it's possible - the backend injects all the necessary parts while preparing index.html, once index.html is served to the browser all the injecting is done already.

Reply:
my point was that this `haveWarnedFactoryFn` was used to avoid logging a warning every time getConfig is called, which is going to be happening many times over the app lifecycle. But since it's memoized, the underlying evaluations are only going to be done once, so this check for `haveWarnedFactoryFn` would only run once as well. It's pretty easy to test - remove it and navigate a few pages in the UI to see if it logs something (you'd probably need to run UI via `yarn start`, not from all-in-one, so that string substitution won't happen).